src/rv/ui/menus/ConnectionMenu.java

```diff
@@ -34,29 +37,119 @@ public ConnectionMenu(Viewer viewer)
 		}
 
 		for (String host : serverHosts) {
-			final JRadioButtonMenuItem item = new JRadioButtonMenuItem(host, getItemCount() == 0);
-			item.addActionListener(e -> SwingUtilities.invokeLater(() -> selectServer(item)));
-			add(item);
+			addHostItem(host, config.getServerPort());
 		}
+
+		add(new JSeparator());
+		JMenuItem m = new JMenuItem("Connect to ...");
+		m.addActionListener((e) -> {
+			int port = config.serverPort; // the default port
+			String message =
+					"<html>Enter an simspark IP address or host name<br><small>You can also specify a port (e.g. 'localhost:4200', 'example.com:4321')</small></html>";
+			String host = JOptionPane.showInputDialog(null, message, "Simspark server", JOptionPane.PLAIN_MESSAGE);
```

Comment:
Passing `null` as the `parentComponent` doesn't work well with multi-monitor setups: the dialog will always appear on the main monitor, even if RoboViz is on a different one.